lib/reek/smells/too_many_constants.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+require_relative 'smell_detector'
+require_relative 'smell_warning'
+
+module Reek
+  module Smells
+    #
+    # A Large Class is a class or module that has a large number of
+    # instance variables, methods, constants or lines of code.
+    #
+    # +TooManyConstants' reports classes having more than a
+    # configurable number of constants.
+    #
+    # See {file:docs/Too-Many-Constants.md} for details.
+    class TooManyConstants < SmellDetector
+      # The name of the config field that sets the maximum number
+      # of constants permitted in a class.
+      MAX_ALLOWED_CONSTS_KEY = 'max_constants'.freeze
+      DEFAULT_MAX_CONSTS = 5
+
+      def self.contexts
+        [:class, :module]
+      end
```

Comment:
If a class is inside the module it will cover. However we might have classes without namespace like:
```
class Test
end
```
And in this case we should look into class context.